packages/workspace/src/command-line/run-tasks/run-many.ts

```diff
@@ -0,0 +1,82 @@
+import * as yargs from 'yargs';
+import { DependencyGraph, ProjectMap, ProjectNode } from '../shared';
+import { runCommand } from './run-command';
+import { getProjects, readEnvironment, splitArgs } from './utils';
+
+export type YargsRunManyOptions = yargs.Arguments & RunManyOptions;
+
+export interface RunManyOptions {
+  target: string;
+  projects: string[];
+  all: boolean;
+  configuration?: string;
+  runner?: string;
+  parallel?: boolean;
+  maxParallel?: number;
+  onlyFailed?: boolean;
+  'only-failed'?: boolean;
+  'max-parallel'?: boolean;
+  verbose?: boolean;
+  help?: boolean;
+  version?: boolean;
+  quiet?: boolean;
+  withDeps?: boolean;
+}
+
+export function runMany(parsedArgs: yargs.Arguments): void {
+  parsedArgs = preprocess(parsedArgs);
+
+  const args = splitArgs(parsedArgs as YargsRunManyOptions, flags);
+  const environment = readEnvironment(args.nxArgs.target);
+
+  let projects = getProjects(
+    args.nxArgs,
+    environment.nxJson,
+    environment.workspaceJson
+  );
+  const graph = projectListToGraph(projects);
+
+  runCommand(projects, graph, args, environment);
+}
+
+function preprocess(args: yargs.Arguments): yargs.Arguments {
+  if (!args.all) {
+    args.all = false;
+  }
+
+  if (!args.projects) {
+    args.projects = [];
+  } else {
+    args.projects = args.projects.split(',');
+  }
+
+  return args;
+}
+
+function projectListToGraph(projects: ProjectNode[]): DependencyGraph {
+  const roots = projects.map(p => p.name);
+  const map: ProjectMap = {};
+  projects.forEach(p => (map[p.name] = p));
+
+  return {
+    projects: map,
```

Comment:
DependencyGraph must contain all projects. This map is only the projects you are running tasks against.